From 5553316729545817e251552aab474d5651fbc222 Mon Sep 17 00:00:00 2001
From: Dirk-Jan Faber <dfaber@victronenergy.com>
Date: Tue, 19 Aug 2025 15:47:24 +0200
Subject: [PATCH 5/5] fix: make dbus text property optional (WIP)

Removes strict validation that filtered out Value-only entries.

This seems to introduce the periodically updating mechanism to
send out more messages than intended.

Fixes #267

(cherry picked from commit 05e5be03bb90e53dce9893214609a31be005a977)
---
 src/services/dbus-listener.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/services/dbus-listener.js b/src/services/dbus-listener.js
index 3881718..1646072 100644
--- a/src/services/dbus-listener.js
+++ b/src/services/dbus-listener.js
@@ -288,7 +288,7 @@ class VictronDbusListener {
         msg.body[0].forEach(entry => {
           const m = { changed: true }
           m.path = entry[0]
-          if (entry[1] && entry[1].length === 2) {
+          if (entry[1]) {
             entry[1].forEach(v => {
               switch (v[0]) {
                 case 'Value': m.value = v[1][1][0]; break
@@ -296,7 +296,7 @@ class VictronDbusListener {
               }
             })
           }
-          if (!m.path || m.value === null || !m.text) {
+          if (!m.path || m.value === null) {
             return
           }
           const service = this.services[msg.sender]
-- 
2.34.1

